app/models/user.rb

```diff
@@ -22,6 +24,10 @@ def active_for_authentication?
 
   private
 
+  def ensure_otp_secret!
+    update!(otp_secret: User.generate_otp_secret) if otp_secret.nil?
```

Comment:
This is happening `before_save` so we don't need to `update!` as we'll make a second DB call; just `self.otp_secret=`